src/pybind/mgr/nfs/export.py

```diff
@@ -880,3 +881,27 @@ def _apply_export(
     def _rados(self, cluster_id: str) -> NFSRados:
         """Return a new NFSRados object for the given cluster id."""
         return NFSRados(self.mgr.rados, cluster_id)
+
+    def get_export_count_with_same_fsal(self, cmount_path: str, cluster_id: str, fs_name: str) -> int:
+        exports = self.list_exports(cluster_id, detailed=True)
+        exports_count = 0
+        for export in exports:
+            if export['fsal']['name'] == 'CEPH' and export['fsal']['cmount_path'] == cmount_path and export['fsal']['fs_name'] == fs_name:
+                exports_count += 1
+        return exports_count
+
+
+def generate_unique_id(cluster_id: str, fs_name: str, cmount_path: str) -> str:
```

Comment:
> so this would be like user1-123, user1-234... it's still creating new user for every export?

no not for every export! only if any field of fsal block changes (fs_name, cmount_path) or cluster_id, we generate new user/key pair